app/models/offspring.rb

```diff
@@ -0,0 +1,10 @@
+class Offspring < ApplicationRecord
+  belongs_to :user
+  self.inheritance_column = :type
+
+  class << self
+    def types
+      %w(AgedOffspring GradedOffspring)
```

Comment:
Why are you doing this? You don't need to define this here. In fact, that is against inheritance, as you would need to change this for another class.